src/components/Patient/PatientInfo/index.tsx

```diff
@@ -18,54 +20,69 @@ import {
   ShowMoreLessButton,
   UserInfo,
 } from '@components/Patient/styles';
-import {
-  cityCountryInfo,
-  emailInfo,
-  genderInfo,
-  lastAppointmentInfo,
-  nameInfo,
-  overviewInformation,
-  phoneNumberInfo,
-  yearsInfo,
-} from '@constants/mockData';
+import { lastAppointmentInfo } from '@constants/mockData';
+import { patientApi } from 'services/PatientService';
+import { male, years } from '@constants/patient';
 
 function PatientCardInfo() {
   const { t } = useTranslation();
   const [showMore, setShowMore] = useState<boolean>(false);
 
   const text = lastAppointmentInfo;
 
+  const { id } = useParams();
+
+  const { data: patient, isLoading } = patientApi.useGetPatientByIdQuery(
+    Number(id)
+  );
+
+  const patientFullName = `${patient?.firstName} ${patient?.lastName}`;
+  const patientCityCountry = `${patient?.city}, ${patient?.country}`;
+  const patientAge = `${
+    new Date().getFullYear() - new Date(patient?.birthDate).getFullYear()
+  } ${years}`;
+
   return (
-    <PatientCardInfoContainer>
-      <PatientInfoName>{nameInfo}</PatientInfoName>
-      <ContactsContainer>
-        <CallIcon />
-        <ContactInfo>{phoneNumberInfo}</ContactInfo>
-        <EmailIcon />
-        <ContactInfo>{emailInfo}</ContactInfo>
-      </ContactsContainer>
-      <InfoContainer>
-        <GenderMaleIcon />
-        <UserInfo>{genderInfo}</UserInfo>
-        <CalendarIcon />
-        <UserInfo>{yearsInfo}</UserInfo>
-        <PinIcon />
-        <UserInfo>{cityCountryInfo}</UserInfo>
-      </InfoContainer>
-      <Overview>
-        <OverviewTitle>{t('Patient.overview')}:</OverviewTitle>
-        {overviewInformation}
-      </Overview>
-      <LastAppointment>
-        <LastAppointmentTitle>
-          {t('Patient.lastAppointment')}
-        </LastAppointmentTitle>
-        {showMore ? text : `${text.substring(0, 250)}...`}
-      </LastAppointment>
-      <ShowMoreLessButton onClick={() => setShowMore(!showMore)}>
-        {showMore ? t('Profile.showLess') : t('Profile.showMore')}
-      </ShowMoreLessButton>
-    </PatientCardInfoContainer>
+    <>
+      {isLoading ? (
+        <div>{t('Dashboard.loading')}</div>
```

Comment:
why loading is not a wrapper, I think you need to have it in many pages

Reply:
fixed